src/Controller/PlanningController.php

```diff
@@ -48,18 +50,54 @@ public function index(Request $request): Response
             'choices' => $this->dataProviderRepository->findAll(),
         ]);
 
+        $form->add('viewType', ChoiceType::class, [
+            'required' => true,
+            'label' => 'planning.view_type',
+            'label_attr' => ['class' => 'label'],
+            'row_attr' => ['class' => 'form-element-wrapper'],
+            'attr' => [
+                'class' => 'form-element',
+            ],
+            'help' => 'planning.data_provider_helptext',
+            'choices' => $this->getTypeChoices(),
+        ]);
+
         $form->handleRequest($request);
 
         if ($form->isSubmitted() && $form->isValid()) {
             $service = $this->dataProviderService->getService($planningFormData->dataProvider);
+            $viewType = $form->getData()->viewType;
+            switch ($viewType) {
+                case 'week':
+                    $planningData = $service->getPlanningDataWeeks();
+
+                    return $this->render('planning/planning-weeks.html.twig', [
```

Comment:
Instead of having 3 almost equal ->render(), you can just set the template according to case

```
$planningData = null;
$template = 'planning/index.html.twig';

if ($form->isSubmitted() && $form->isValid()) {
  $viewType = $form->getData()->viewType;
    switch ($viewType) {
        case 'week':
            $planningData = $service->getPlanningDataWeeks();
            $template = 'planning/planning-weeks.html.twig';
        ...

         return $this->render($template, [
            'controller_name' => 'PlanningController',
            'planningData' => $planningData,
            'form' => $form,
        ]);

```